nautobot/dcim/models/device_components.py

```diff
@@ -773,6 +778,87 @@ def parent(self):
         return self.device
 
 
+@extras_features(
+    "custom_fields",
+    "custom_links",
+    "custom_validators",
+    "export_templates",
+    "graphql",
+    "relationships",
+    "statuses",
+    "webhooks",
+)
+class InterfaceRedundancyGroup(PrimaryModel, ConfigContextModel, StatusModel):  # pylint: disable=too-many-ancestors
+    """
+    A collection of Interfaces that supply a redundancy group for protocols like HSRP/VRRP.
+    """
+
+    name = models.CharField(max_length=100, unique=True)
+    slug = AutoSlugField(populate_from="name")
+    description = models.CharField(max_length=200, blank=True)
+    virtual_ip = models.ForeignKey(
+        to="ipam.IPAddress",
+        on_delete=models.CASCADE,
+        null=True,
+        related_name="interface_redundancy_virtual_ip",
+    )
+    interfaces = models.ManyToManyField(
+        to="dcim.Interface",
+        through="dcim.InterfaceRedundancyGroupAssociation",
+        related_name="redundancy_groups",
+        blank=True,
+    )
+    protocol = models.CharField(
+        max_length=50,
+        blank=True,
+        choices=InterfaceRedundancyGroupProtocolChoices,
+        verbose_name="Redundancy Protocol",
+    )
+    secrets_group = models.ForeignKey(
+        to="extras.SecretsGroup",
+        on_delete=models.SET_NULL,
+        default=None,
+        blank=True,
+        null=True,
+    )
+    group_id = models.CharField(max_length=50, blank=True)
+
+    class Meta:
+        """Meta class."""
+
+        ordering = ["name"]
+
+    def get_absolute_url(self):
+        """Return detail view for InterfaceRedundancyGroup."""
+        return reverse("dcim:interfaceredundancygroup", args=[self.id])
+
+    def __str__(self):
+        """Return a string representation of the instance."""
+        return self.name
+
+
+@extras_features(
+    "relationships",
+    "custom_fields",
+)
+class InterfaceRedundancyGroupAssociation(PrimaryModel):
+    """Intermediary model for associating Interface(s) to InterfaceRedundancyGroup(s)."""
+
+    group = models.ForeignKey(to="dcim.InterfaceRedundancyGroup", on_delete=models.CASCADE)
+    interface = models.ForeignKey(to="dcim.Interface", on_delete=models.CASCADE)
+    priority = models.PositiveSmallIntegerField()
```

Comment:
If we feel we desire an easier way to pull out the `priority` value from these assignments we likely want to add a property on the `InterfaceRedundancyGroup` model.

Reply:
Sorry if this came across as terse. Seeing as there would be more than one `priority` (one per related `interface`), this comment can be disregarded. :)